### PR TITLE
DHFPROD-7270: Add structured properties to the test matched result table

### PR DESCRIPTION
**Pull request by AkshayBajajML**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by AkshayBajajML** [comment]
@jbelonoj approved UI changes!

File: marklogic-data-hub-central/ui/src/components/entities/matching/matching-step-detail/matching-step-detail.tsx [hunk 1]
```diff
@@ -159,6 +160,13 @@ const MatchingStepDetail: React.FC = () => {
     await test();
     setPreviewMatchedActivity(previewMatchActivity);
     setRulesetDataList(rulesetDataList);
+    let response = await axios.get("/api/steps/matching");
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/entities/matching/matching-step-detail/matching-step-detail.tsx:163]: Let's put all API calls under an appropriate location within **/src/api/***.
This seems like a good candidate for "getSteps", under **/src/api/steps.tsx** and then we can simply call it here like **getSteps("matching")**
Also, let's add another layer of validation by checking if you received a proper response before actually using it further. One of the ways is to check if the value of response.status is "200".

File: marklogic-data-hub-central/ui/src/components/entities/matching/expandable-table-view/expandable-table-view.tsx [hunk 2]
```diff
@@ -29,7 +32,7 @@ const testMatchedUriTableColumns = [
     key: "matchedRulesetType " + (counter++) + " exact",
     width: "6%",
     render: (matchedRulesetType, key, index) => (matchedRulesetType.map(rulesetType => {
-      return (rulesetType === "Exact") && <span className={styles.testMatchedColumns} key={key} aria-label={matchedRulesetType + " " + (index)}>
+      return (rulesetType === "exact" || rulesetType === "Exact") && <span className={styles.testMatchedColumns} key={key} aria-label={matchedRulesetType + " " + (index)}>
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/entities/matching/expandable-table-view/expandable-table-view.tsx:35]: If the rulesetType is going to be checked for different variations of the same word then might as well convert it to upper or lower case and check it only once.
something like **rulesetType.toLowerCase() === "exact"**
Same goes for all other places where rulesetType is being checked.

File: marklogic-data-hub-central/ui/src/components/entities/matching/expandable-table-view/expandable-table-view.tsx [hunk 3]
```diff
@@ -95,36 +98,102 @@ const testMatchedUriTableColumns = [
...
     width: "15%",
     render: (scores, key) =>  <span key={key}  aria-label={"score " + scores.scores[0]}>
-      <Progress percent={scores.scores[0]} strokeWidth={20} strokeColor={scores.matchedRule[0] !== "Reduce" ? "#00b300" : "#ff0000"} format={scores.matchedRule[0] !== "Reduce" ? percent => `${percent}` : percent => `-${percent}`} strokeLinecap={"square"}/>
+      {scores.scores[0]>0 && <Progress percent={scores.scores[0]} strokeWidth={20} strokeColor={scores.matchedRule[0] !== "Reduce" ? "#00b300" : "#ff0000"} format={scores.matchedRule[0] !== "Reduce" ? percent => `${percent}` : percent => `-${percent}`} strokeLinecap={"square"}/>}
     </span>
   }
 ];
 
+const getMatchingData = async () => {
+  let response = await axios.get("/api/steps/matching");
```
Comment (xnikhil08) [inline marklogic-data-hub-central/ui/src/components/entities/matching/expandable-table-view/expandable-table-view.tsx:107]: Same comment as in another place. Let's move all the API calls to /src/api/*.

**Review by xnikhil08 — CHANGES_REQUESTED** [review]

**Comment by AkshayBajajML** [comment]
@xnikhil08 thanks for the feedback; updated the code and pushed the changes again.

**Review by xnikhil08 — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/entities/matching/matching-step-detail/matching-step-detail.tsx [hunk 4]
```diff
@@ -151,14 +152,30 @@ const MatchingStepDetail: React.FC = () => {
...
+        let entityName = curationOptions.activeStep.entityName;
+        for (let i = 0; i < response.data.length; i++) {
+          if (response.data[i].entityType === entityName) {
+            setMatchedRulesetData(response.data[i]);
+          }
+        }
+      }
+    } catch (error) {
+      let message = error;
+      console.error("Error While fetching the mapping data", message);
```
[the remark this answers is not in the document]
Comment (AkshayBajajML) [reply marklogic-data-hub-central/ui/src/components/entities/matching/matching-step-detail/matching-step-detail.tsx]: Done!!

**Review by xnikhil08 — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**